test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -126,25 +122,18 @@ class OverrideHostTest : public XdsEnd2endTest {
     return listener;
   }
 
-  // Send requests until a desired backend is hit and returns cookie name/value
-  // pairs. Empty collection is returned if the backend was never hit.
-  // For weighted clusters, more than one request per backend may be necessary
-  // to obtain the cookie. max_requests_per_backend argument specifies
-  // the number of requests per backend to send.
-  std::vector<std::pair<std::string, std::string>>
-  GetAffinityCookieHeaderForBackend(
+  std::vector<Cookie> GetCookiesForBackend(
```

Comment:
Instead of having this return a vector, you could pass in the cookie name and have it return only the one you're actually interested in.

Reply:
I wanted to be able to assert in the test that there's no cookie under a different name. E.g. there should be no cookie with an original name when route config overwrites the name. Filtering by name would mean this method returns an `optional`, so code will not be more that much simpler.